Source/swiftlint/Extensions/Array+SwiftLint.swift

```diff
@@ -6,12 +6,14 @@ extension Array {
     }
 
     func parallelMap<T>(transform: (Element) -> T) -> [T] {
-        var result = ContiguousArray<T?>(repeating: nil, count: count)
-        return result.withUnsafeMutableBufferPointer { buffer in
-            DispatchQueue.concurrentPerform(iterations: buffer.count) { idx in
-                buffer[idx] = transform(self[idx])
+        return [T](unsafeUninitializedCapacity: count) { buffer, initializedCount in
+            let baseAddress = buffer.baseAddress!
```

Comment:
Is this always guaranteed to succeed? Is that behavior documented?

Reply:
In this case yes. Documented, not really. That's how I ran into this: https://bugs.swift.org/browse/SR-14164

If you follow throw this, the buffer is initialized with a non-optional baseAddress: https://github.com/apple/swift/blob/06dc593cece2ded7457ba3d5a6c41857881da157/stdlib/public/core/Array.swift#L1468